### PR TITLE
[CHK-1071] Feature/add profile factory

### PR DESCRIPTION
**Pull request by ArthurSampaio**
#### What problem is this solving?
Add factory function to create providers in different contexts

#### How should this be manually tested?
You must go to the [ws](https://controllers--checkoutio.myvtex.com/) and should be able to complete an order like before. 

[Workspace](url)

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Clubhouse story (if applicable).
- [ x] Updated/created tests (important for bug fixes).
- [] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage
No needed

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,8 +24,8 @@
     "@types/node": "^13.7.7",
     "@vtex/prettier-config": "^0.1.3",
     "eslint": "^6.8.0",
-    "eslint-config-vtex": "^12.2.1",
-    "eslint-config-vtex-react": "^6.2.1",
+    "eslint-config-vtex": "^12.0.3",
```
Comment (ArthurSampaio) [inline package.json:27]: need to keep consistency between the controllers. In the near future, we can update it again.

File: react/OrderProfile.tsx [hunk 2]
```diff
@@ -1,142 +1,91 @@
...
 
-const SET_PROFILE_TASK = 'SetProfileTask'
-const SET_CLIENT_PREFERENCES_TASK = 'SetClientPreferencesTask'
-
-export const OrderProfileProvider: React.FC = ({ children }) => {
-  const { enqueue, listen } = useOrderQueue()
-  const { setOrderForm } = useOrderForm()
-
-  const queueStatusRef = useQueueStatus(listen)
+// const { useQueueStatus } = OrderQueue
```
Comment (ArthurSampaio) [inline react/OrderProfile.tsx]: 🔨

Comment (wdsrocha) [reply react/OrderProfile.tsx]: wait, what does that mean?

**Review by lucasecdb — DISMISSED** [review]
lgtm, just remove the commented code and you are good to go :-)

File: react/__tests__/createOrderProfileProvider.test.tsx [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+} from '../components/createOrderProfileProvider'
+import { LogParams } from '../utils/logger'
+
+const mockProfileData = {
+  firstName: 'Sicrano ',
+  lastName: 'DiTal',
+  phone: '+55(12) 3 1231-223',
+  documentType: 'cpf',
+  document: '123.456.789-09',
+  email: 'fulano@detal.com',
```
Comment (jeffersontuc) [inline react/__tests__/createOrderProfileProvider.test.tsx:24]: 😆

**Review by jeffersontuc — DISMISSED** [review]
👌🏾

File: react/OrderProfile.tsx (hunk 2 above)
Comment (ArthurSampaio) [reply react/OrderProfile.tsx]: Is my way to say "hey fix this".

**Review by jeffersontuc — APPROVED** [review]

**Review by lucasecdb — APPROVED** [review]

**Pull request merged**